webpack/assets/javascripts/react_app/components/ReportTemplate/ReportTemplateGenerate.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { OverlayTrigger, Tooltip, Icon } from 'patternfly-react';
+import DateTime from './components/ReportDateTime';
+import Autocomplete from './components/ReportAutoComplete';
+import { translate as __ } from '../../common/I18n';
+
+const TYPES = { PLAIN: 'plain', DATE: 'date', SEARCH: 'search' };
+const ReportTemplateGenerate = ({
```

Comment:
Isn't this name bit missleading? `ReportInput`/`TemplateInput` sounds bit better to me.